src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -248,8 +243,8 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
 
         // Transform the effectiveRoles from a Set to a List.
         return effectiveRoles.entrySet().stream()
-            .map(x -> new AbstractMap.SimpleEntry<>(x.getKey(), new ArrayList<>(x.getValue())))
-            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            .map(roleSet -> new AbstractMap.SimpleEntry<>(roleSet.getKey(), new ArrayList<>(roleSet.getValue())))
```

Comment:
There is no special need for a `List` vs. a `Set` in the signature derived from `AccessRolesProvider.findRolesForPath`:
https://github.com/fcrepo4/fcrepo-module-auth-rbacl/blob/master/fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AccessRolesProvider.java#L75

..as well as `AccessRolesProvider.getRoles`:
https://github.com/fcrepo4/fcrepo-module-auth-rbacl/blob/master/fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AccessRolesProvider.java#L47

That is just how it was initially implemented... and propagated through RBACL, XACML, and WebAC.